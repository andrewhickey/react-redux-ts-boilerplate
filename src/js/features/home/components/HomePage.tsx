import * as React from "react"
import { withProps } from "recompose"

export interface IHomeProps {
  name: string
  number: number
}

export const HomePageComponent = ({ name }: IHomeProps) => (
  <div>Hello dog {name}</div>
)

export const homeWrapper1 = withProps({ name: "dog" })
export const homeWrapper2 = withProps({ number: 2 })

export default homeWrapper2(homeWrapper1(HomePageComponent))
