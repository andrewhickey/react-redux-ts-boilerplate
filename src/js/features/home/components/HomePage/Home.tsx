import * as React from 'react';
import { compose, withProps, ComponentEnhancer } from 'recompose';

export interface HomeProps {
  name: string;
  number: number;
};

export const HomePage = ({ name }: HomeProps) => (
  <div>Hello dog {name}</div>
);

export const homeWrapper1 = withProps({ name: 'dog'})
export const homeWrapper2 = withProps({ number: 2 })

export const Home =  homeWrapper2(homeWrapper1(HomePage));
