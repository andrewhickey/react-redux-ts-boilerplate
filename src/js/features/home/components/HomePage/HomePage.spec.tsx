import * as React from "react"
import * as ReactDOM from "react-dom"
import * as ReactTestRenderer from "react-test-renderer"
import { HomePage, HomeProps, Home } from "./HomePage"

it("renders correctly with no HoC", () => {
  const props: HomeProps = { name: "test", number: 1 }
  const tree = ReactTestRenderer.create(<HomePage {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders correctly with the HoC", () => {
  const tree = ReactTestRenderer.create(<Home />).toJSON()
  expect(tree).toMatchSnapshot()
})
