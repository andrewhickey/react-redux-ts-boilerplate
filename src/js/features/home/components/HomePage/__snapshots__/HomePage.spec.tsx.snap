// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no HoC 1`] = `
<div>
  Hello dog 
  test
</div>
`;

exports[`renders correctly with the HoC 1`] = `
<div>
  Hello dog 
  dog
</div>
`;
